dom.go

```diff
@@ -1946,120 +1992,196 @@ func (e *HTMLCanvasElement) GetContext(param string) *js.Object {
+
+func (ctx *CanvasRenderingContext2D) FillText(text string, x, y, maxWidth float64) {
+	if maxWidth == -1 {
+		ctx.Call("fillText", text, x, y)
+		return
+	}
+
+	ctx.Call("fillText", text, x, y, maxWidth)
 }
 
-func (ctx *CanvasRenderingContext2D) StrokeRect(x, y, width, height int) {
-	ctx.Call("strokeRect", x, y, width, height)
+func (ctx *CanvasRenderingContext2D) StrokeText(text string, x, y, maxWidth float64) {
+	if maxWidth == -1 {
+		ctx.Call("strokeText", text, x, y)
+		return
+	}
+
+	ctx.Call("strokeText", text, x, y, maxWidth)
+}
+func (ctx *CanvasRenderingContext2D) MeasureText(text string) *TextMetrics {
+	textMetrics := ctx.Call("measureText", text)
+	return &TextMetrics{Object: textMetrics}
 }
 
-func (ctx *CanvasRenderingContext2D) ClearRect(x, y, width, height int) {
-	ctx.Call("clearRect", x, y, width, height)
+// Line styles
+
+func (ctx *CanvasRenderingContext2D) GetLineDash() interface{} {
+	return ctx.Call("getLineDash")
 }
 
-// Paths
+//
+func (ctx *CanvasRenderingContext2D) SetLineDash(arr interface{}) {
+	ctx.Call("setLineDash", arr)
+	return
+}
 
-func (ctx *CanvasRenderingContext2D) Fill() {
-	ctx.Call("fill")
+// Gradients and patterns
+
+func (ctx *CanvasRenderingContext2D) CreateLinearGradient(x0, y0, x1, y1 float64) {
+	ctx.Call("createLinearGradient", x0, y0, x1, y1)
 }
 
-func (ctx *CanvasRenderingContext2D) Stroke() {
-	ctx.Call("stroke")
+func (ctx *CanvasRenderingContext2D) CreateRadialGradient(x0, y0, r0, x1, y1, r1 float64) {
+	ctx.Call("createRadialGradient", x0, y0, r0, x1, y1, r1)
 }
 
-func (ctx *CanvasRenderingContext2D) BeginPath() {
-	ctx.Call("beginPath")
+func (ctx *CanvasRenderingContext2D) CreatePattern(image *Element, repetition string) {
+	ctx.Call("createPattern", image, repetition)
 }
 
-func (ctx *CanvasRenderingContext2D) MoveTo(x, y int) {
-	ctx.Call("moveTo", x, y)
+// Paths
+
+func (ctx *CanvasRenderingContext2D) BeginPath() {
+	ctx.Call("beginPath")
 }
 
 func (ctx *CanvasRenderingContext2D) ClosePath() {
 	ctx.Call("closePath")
 }
 
+func (ctx *CanvasRenderingContext2D) MoveTo(x, y int) {
+	ctx.Call("moveTo", x, y)
+}
+
 func (ctx *CanvasRenderingContext2D) LineTo(x, y int) {
 	ctx.Call("lineTo", x, y)
 }
 
-func (ctx *CanvasRenderingContext2D) Clip() {
-	ctx.Call("clip")
+func (ctx *CanvasRenderingContext2D) BezierCurveTo(cp1x, cp1y, cp2x, cp2y, x, y int) {
+	ctx.Call("bezierCurveTo", cp1x, cp1y, cp2x, cp2y, x, y)
 }
 
-func (ctx *CanvasRenderingContext2D) QuadraticCurveTo(cpx, cpy, x, y int) {
+func (ctx *CanvasRenderingContext2D) QuadraticCurveTo(cpx, cpy, x, y float64) {
 	ctx.Call("quadraticCurveTo", cpx, cpy, x, y)
 }
 
-func (ctx *CanvasRenderingContext2D) BezierCurveTo(cp1x, cp1y, cp2x, cp2y, x, y int) {
-	ctx.Call("bezierCurveTo", cp1x, cp1y, cp2x, cp2y, x, y)
-}
-
-func (ctx *CanvasRenderingContext2D) Arc(x, y, r int, sAngle, eAngle float64, counterclockwise bool) {
+func (ctx *CanvasRenderingContext2D) Arc(x, y, r, sAngle, eAngle float64, counterclockwise bool) {
```

Comment:
Have you verified that `x`, `y`, and `r` all can be floating point values, making the `int` -> `float64` breaking API change neccessary?

Reply:
I changed in CreateImageData the type from float64 to int.

Reply:
What about `GetImageData`, `PutImageData` and `PutImageDataDirty`, some of those also involve widths/heights. Should they be changed to `int`? If not, what's the reason to keep them `float64`?

Reply:
Ok i've checked all three, ```GetImageData``` doesn't suffer any changes and works with floating values,
```PutImageData``` does work with floats and it has a graphical difference (same goes with ```PutImageDataDirty``` for all of the parameters).


Also to put this question to an end: https://stackoverflow.com/questions/29371695/does-canvas-support-floating-point-numbers-when-drawing-line-or-arc

I also noticed i left ```LineTo``` ```MoveTo``` ```BezierCurveTo``` with int, i changed them to float64 as those work aswell with floating values. (I verified that)

Reply:
Another reason would be ease of use, i agree with CreateImageData that it's a good idea to use int,
(like ease of use of len as an input).
All the others do work with float64 and also it makes more sense to keep them with only one datatype.

So i think this is now done, unless you have some other doubts or issues you would like to address.